### ISSUE TITLE
Performance Issue with cron Job

### ISSUE THREAD

**Issue by sudarshann** [Mowdirect/stock_module#10]
**Describe the bug**
Optimize the cron job handler to use low resources.
https://cl.ly/9d653dc62f59

**To Reproduce**
1. Observe the CPU usage during Cron Job run using Blackfire.io or equalent

**Expected behavior**
CPU should not spike high during cron job run

**Screenshots**
![image](https://user-images.githubusercontent.com/424396/64510156-14dc8300-d2ff-11e9-85f8-f97831e86d0d.png)

**Additional context**
https://labs.octivi.com/10-worst-magento-practices/
And a link saying more or less the same things that might be easier to distribute to your devs. 
Basically Magento made several poor design decisions. They hit the database hard when loading models. So you need to avoid inside of loops is called inside a loop

https://alanstorm.com/magento_collections/ will teach you how to extend a collection to add the data you need to it

another good article on db manipulation Magento style https://www.techdilate.com/magento-database-queries/
